src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -251,6 +254,37 @@ public RootNode doGetStudyTree(UUID studyId) {
         return root;
     }
 
+    @Transactional
+    public void copyStudyTree(AbstractNode nodeToDuplicate, UUID nodeParentId, UUID studyId) {
+        UUID rootId = null;
+        if (NodeType.ROOT.equals(nodeToDuplicate.getType())) {
+            StudyEntity tempStudy = new StudyEntity();
```

Comment:
I think you should pass, as an argument to the copyStudyTree method, the new StudyEntity returned by insertStudy called before in the createStudy method